src/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/CmsSignature.ECDsa.cs

```diff
@@ -65,8 +65,13 @@ protected override bool VerifyKeyType(AsymmetricAlgorithm key)
                     return false;
                 }
 
-                // 2 * KeySizeBytes => 2 * KeySizeBits / 8 => KeySizeBits / 4
-                int bufSize = key.KeySize / 4;
+                int bufSize;
+                checked
+                {
+                    // fieldSize = ceil(KeySizeBits / 8);
+                    int fieldSize = (key.KeySize + 7) / 8;
+                    bufSize = 2 * fieldSize;
```

Comment:
Math says that the multiply by 2 can't overflow given that it was preceded by a successful divide by 8, but that's fine :)